bot/exts/utils/thread_bumper.py

```diff
@@ -0,0 +1,112 @@
+
+            try:
+                thread = await self.bot.fetch_channel(thread_id)
+            except discord.NotFound:
+                log.info(f"Thread {thread_id} has been deleted, removing from bumped threads.")
+                await self.threads_to_bump.delete(thread_id)
+            if thread.archived:
+                await thread.edit(archived=False)
+
+    @commands.group(name="bump")
+    async def thread_bump_group(self, ctx: commands.Context) -> None:
+        """A group of commands to manage the bumping of threads."""
+        if not ctx.invoked_subcommand:
+            await ctx.send_help(ctx.command)
+
+    @thread_bump_group.command(name="add")
+    async def add_thread_to_bump_list(self, ctx: commands.Context, thread_id: t.Optional[int]) -> None:
+        """Add a thread to the bump list."""
+        await self.init_task
+
+        if not thread_id and isinstance(ctx.channel, discord.Thread):
+            thread_id = ctx.channel.id
+        else:
+            raise commands.BadArgument("You must provide a thread id, or run this command in a thread.")
+
+        await self.threads_to_bump.set(thread_id, "sentinel")
+        await ctx.send(f":ok_hand: Thread <#{thread_id}> has been added to the bump list.")
+
+    @thread_bump_group.command(name="remove", aliases=("r", "rem", "d", "del", "delete"))
+    async def remove_thread_from_bump_list(self, ctx: commands.Context, thread_id: t.Optional[int]) -> None:
+        """Remove a thread from the bump list."""
+        await self.init_task
+
+        if not thread_id and isinstance(ctx.channel, discord.Thread):
+            thread_id = ctx.channel.id
+        else:
+            raise commands.BadArgument("You must provide a thread id, or run this command in a thread.")
+
+        await self.threads_to_bump.delete(thread_id)
+        await ctx.send(f":ok_hand: Thread <#{thread_id}> has been removed from the bump list.")
+
+    @thread_bump_group.command(name="list", aliases=("get",))
+    async def list_all_threads_in_bump_list(self, ctx: commands.Context) -> None:
+        """List all the threads in the bump list."""
+        await self.init_task
+
+        lines = [f"<#{k}>" for k, _ in await self.threads_to_bump.items()]
+        embed = discord.Embed(
+            title="Threads in the bump list",
+            colour=constants.Colours.blue
+        )
+        await LinePaginator.paginate(lines, ctx, embed)
+
+    @commands.Cog.listener()
+    async def on_thread_update(self, _: discord.Thread, after: discord.Thread) -> None:
+        """
+        Listen for thread updates and check if the thread has been archived.
+
+        If the thread has been archived, and is in the bump list, un-archive it.
+        """
+        await self.init_task
+
+        if not after.archived:
+            return
+
+        bumped_threads = [k for k, _ in await self.threads_to_bump.items()]
+        if after.id in bumped_threads:
+            await after.edit(archived=False)
```

Comment:
Would manually archived threads be bumped too? I don't think that'd be a desired behaviour.
[Thread.archiver_id](<https://discordpy.readthedocs.io/en/master/api.html?highlight=on_thread_update#discord.Thread.archiver_id>) is `Optional[int]`. I'm assuming it will not be `None` in cases of manual archiving. I've not tested it but I guess you could use that to check if a thread was manually archived.

Reply:
Please correct me if I'm wrong. As perfect as I might seem, I make sometimes make mistakes.

Reply:
Hmmm, I've tried this in the on_thread_update event listener, and it doesn't seem to be set at all, on the before or after thread object when manually archiving.

i also can't see where this data would come from, since it's not present on the discord api https://discord.com/developers/docs/resources/channel#thread-metadata-object and in the discord.py source, its tries to pull this key directly form the data from the api.

Reply:
So possibly it used to be there, but isn't anymore? 🤷

Reply:
I've been bamboozled!

> As of June 10th, 2021, this value will always be None since Discord doesn’t provide this information anymore.
This property will be removed in a future version.

[(From Disnake docs)](<https://docs.disnake.dev/en/latest/api.html?highlight=on_thread_update#disnake.Thread.archiver_id>)

I guess we can manually remove the thread from the bump list before archiving it.

Reply:
Ah, that explains it.

The only other option I can think of is to take the last message in the thread, add the auto-archive time to it and see if that is close to the current time, or in the past.

Reply:
Done in https://github.com/python-discord/bot/pull/2084/commits/2d8fa806aff55e5efdbd6b74440acf78c256fdd5

Reply:
Hi, saw this and wanted to explain a 100% guaranteed way to know if it was manual or automatic: manual closes will have an audit log entry. If there's no relevant audit log entry then it safe to assume that the thread closed automatically, rather than manual.

Reply:
> Hi, saw this and wanted to explain a 100% guaranteed way to know if it was manual or automatic: manual closes will have an audit log entry. If there's no relevant audit log entry then it safe to assume that the thread closed automatically, rather than manual.

Were you expecting something like this?
https://github.com/python-discord/bot/pull/2084/commits/0ae5ee9c478bd4a627473acce8a46de3965176b9